crates/bevy_ecs/src/component.rs

```diff
@@ -2115,8 +2122,20 @@ impl RequiredComponents {
     // Merges `required_components` into this collection. This only inserts a required component
     // if it _did not already exist_.
     pub(crate) fn merge(&mut self, required_components: &RequiredComponents) {
```

Comment:
Nit: This comment could now be updated to clarify that if a required component *does* already exist, its constructor may still get updated *if* the new requirement is more specific (meaning that the inheritance depth is smaller).

Also this should probably be a doc comment (`///`) instead of a regular comment.